src/Native/Unix/CMakeLists.txt

```diff
@@ -22,7 +22,10 @@ add_compile_options(-fPIC)
 add_compile_options(-I${CMAKE_CURRENT_SOURCE_DIR}/Common)
 add_compile_options(-I${CMAKE_CURRENT_BINARY_DIR}/Common)
 add_compile_options(-g)
-add_compile_options(-Wthread-safety)
+if(CMAKE_C_COMPILER_ID STREQUAL Clang)
+    add_compile_options(-Wthread-safety)
+    add_compile_options(-Wno-thread-safety-analysis)
```

Comment:
@jkotas, @stephentoub, @janvorli 

This (`-Wno-thread-safety-analysis`) option was added to suppress the following error I received from Clang 6.0 on FreeBSD 11.2 (gcc doesn't complain on the same OS or clang on others; Linux / Mac):

```sh
[ 51%] Building C object System.Native/CMakeFiles/System.Native-Static.dir/pal_uid.c.o
/usr/home/am11/corefx/src/Native/Unix/System.Native/pal_uid.c:138:13: error: mutex 's_groupLock' is not held on every path through here [-Werror,-Wthread-safety-analysis]
            break;
            ^
/usr/home/am11/corefx/src/Native/Unix/System.Native/pal_uid.c:133:14: note: mutex acquired here
        rv = pthread_mutex_lock(&s_groupLock);
             ^
/usr/home/am11/corefx/src/Native/Unix/System.Native/pal_uid.c:116:1: note: Thread warning in function 'SystemNative_GetGroupList'
{
^
1 error generated.
*** [System.Native/CMakeFiles/System.Native-Static.dir/pal_uid.c.o] Error code 1

make[2]: stopped in /usr/home/am11/corefx/artifacts/obj/native/FreeBSD-x64-Debug
1 error

make[2]: stopped in /usr/home/am11/corefx/artifacts/obj/native/FreeBSD-x64-Debug
*** [System.Native/CMakeFiles/System.Native-Static.dir/all] Error code 2

make[1]: stopped in /usr/home/am11/corefx/artifacts/obj/native/FreeBSD-x64-Debug
1 error

make[1]: stopped in /usr/home/am11/corefx/artifacts/obj/native/FreeBSD-x64-Debug
*** [all] Error code 2

make: stopped in /usr/home/am11/corefx/artifacts/obj/native/FreeBSD-x64-Debug
1 error

make: stopped in /usr/home/am11/corefx/artifacts/obj/native/FreeBSD-x64-Debug
Failed to build corefx native components.
```

Should I create an issue for it?

Reply:
I do not see any real bug here. Looks like bogus compiler warning to me.